### DISCUSSION TITLE
panic: runtime error: slice bounds out of range [:23044] with capacity 20480

### DISCUSSION THREAD

**Discussion by fperfeito** [sijms/go-ora#94] [Q&A] [answered]
Use version 2.1.23

Table: 
   table_name_XML
columns: 
name		data_type	nullable
id_code         number(10) 	No
TX_XML	        CLOB		Yes		 	
TX_XML_RETURN	CLOB		Yes		 	
DT_POST		DATE		No

code:

	now := time.Now()
	minutoAnt := now.Add(-1 * time.Minute)
	I := time.Date(minutoAnt.Year(), minutoAnt.Month(), minutoAnt.Day(), minutoAnt.Hour(), minutoAnt.Minute(), 0, 0, time.UTC).Format("2006-01-02 15:04:05")
	F := time.Date(minutoAnt.Year(), minutoAnt.Month(), minutoAnt.Day(), minutoAnt.Hour(), minutoAnt.Minute(), 59, 999999999, time.UTC).Format("2006-01-02 15:04:05")

	query := SELECT id_code, TX_XML as TX_XML FROM table_name_XML WHERE DT_POST BETWEEN to_date(:I, 'yyyy/mm/dd hh24:mi:ss') AND to_date(:F, 'yyyy/mm/dd hh24:mi:ss')

	rows, err := connOra.Query(query, I, F)
	if err != nil {
		...
	}
	defer rows.Close()
	for rows.Next() {
		err := rows.Scan(&code, &txXML)
		if err != nil {
			...
		}
		...
	}

ERROR:

panic: runtime error: slice bounds out of range [:23044] with capacity 20480

goroutine 30 [running]:
github.com/sijms/go-ora/v2/network.(*Session).read(0xc0000ac000, 0x59ff, 0x0, 0x0, 0x0, 0x0, 0x0)
	D:/Users/go/pkg/mod/github.com/sijms/go-ora/v2@v2.1.23/network/session.go:295 +0x54a
github.com/sijms/go-ora/v2/network.(*Session).GetBytes(0xc0000ac000, 0x59ff, 0x0, 0x0, 0x0, 0x0, 0x0)
	D:/Users/go/pkg/mod/github.com/sijms/go-ora/v2@v2.1.23/network/session.go:864 +0x8f
github.com/sijms/go-ora/v2.(*Lob).readData(0xc0001ae708, 0xc0000ac000, 0x0, 0x0)
	D:/Users/go/pkg/mod/github.com/sijms/go-ora/v2@v2.1.23/lob.go:281 +0x4cf
github.com/sijms/go-ora/v2.(*Lob).read(0xc0001ae708, 0xc0000a0000, 0x0, 0x0)
	D:/Users/go/pkg/mod/github.com/sijms/go-ora/v2@v2.1.23/lob.go:210 +0x939
github.com/sijms/go-ora/v2.(*Lob).getData(0xc0001ae708, 0xc0000a0000, 0x0, 0x0, 0x0, 0x0, 0x0)
	D:/Users/go/pkg/mod/github.com/sijms/go-ora/v2@v2.1.23/lob.go:56 +0x16e
github.com/sijms/go-ora/v2.(*defaultStmt).read(0xc00017c280, 0xc000168280, 0x0, 0x0)
	D:/Users/go/pkg/mod/github.com/sijms/go-ora/v2@v2.1.23/command.go:729 +0x1d14
github.com/sijms/go-ora/v2.(*Stmt).Query(0xc00017c280, 0xc00005ef80, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0)
	D:/Users/go/pkg/mod/github.com/sijms/go-ora/v2@v2.1.23/command.go:1240 +0x5b8
database/sql.ctxDriverStmtQuery(0x1449f90, 0xc0000180a0, 0x144a118, 0xc00017c280, 0xc00006a9b0, 0x2, 0x2, 0x0, 0x0, 0x0, ...)
	C:/Program Files/Go/src/database/sql/ctxutil.go:94 +0x435
database/sql.rowsiFromStatement(0x1449f90, 0xc0000180a0, 0x1449520, 0xc0000a0000, 0xc000068dc0, 0xc000073698, 0x2, 0x2, 0x0, 0x0, ...)
	C:/Program Files/Go/src/database/sql/sql.go:2695 +0x2b2
database/sql.(*DB).queryDC(0xc00002f1e0, 0x1449f90, 0xc0000180a0, 0x0, 0x0, 0xc0000bc090, 0xc00005d740, 0x14016c1, 0xbc, 0xc000073698, ...)
	C:/Program Files/Go/src/database/sql/sql.go:1680 +0x8c9
database/sql.(*DB).query(0xc00002f1e0, 0x1449f90, 0xc0000180a0, 0x14016c1, 0xbc, 0xc000073698, 0x2, 0x2, 0x13e7001, 0x0, ...)
	C:/Program Files/Go/src/database/sql/sql.go:1628 +0x228
database/sql.(*DB).QueryContext(0xc00002f1e0, 0x1449f90, 0xc0000180a0, 0x14016c1, 0xbc, 0xc0001af698, 0x2, 0x2, 0x0, 0x0, ...)
	C:/Program Files/Go/src/database/sql/sql.go:1605 +0xfe
database/sql.(*DB).Query(0xc00002f1e0, 0x14016c1, 0xbc, 0xc000073698, 0x2, 0x2, 0x0, 0x0, 0x0)
	C:/Program Files/Go/src/database/sql/sql.go:1619 +0xe5

to simulate the error, you need to create 100 records in the informed table and the XML fields must contain more than 5000 characters in each field and each record.

the error happens after line 775 of the command.go file and then the buffer growth. the error occurs after reading a number of rows from the table

DEBUG:

func (session *Session) read(numBytes int) ([]byte, error) {
	if session.index+numBytes > len(session.inBuffer) {
		pck, err := session.readPacket()
		if err != nil {
			return nil, err
		}
		if dataPck, ok := pck.(*DataPacket); ok {
			session.inBuffer = append(session.inBuffer, dataPck.buffer...)
		} else {
			return nil, errors.New("the packet received is not data packet")
		}
	}
	fmt.Println("session.index ", session.index)
	fmt.Println("numBytes ", numBytes)
	fmt.Println("session.index+numBytes ", session.index+numBytes)
	if session.index+numBytes > 32000 {
		fmt.Println("erro")
	}
	ret := session.inBuffer[session.index : session.index+numBytes]
	fmt.Println("ret ", ret)
	session.index += numBytes
	fmt.Println("session.index ", session.index)
	return ret, nil
}

session.index  150
numBytes  1
session.index+numBytes  151
ret  [0]
session.index  151
session.index  151
numBytes  1
session.index+numBytes  152
ret  [0]
session.index  152
session.index  152
numBytes  1
session.index+numBytes  153
ret  [0]
session.index  153
session.index  0
numBytes  1
session.index+numBytes  1
ret  [14]
session.index  1
session.index  1
numBytes  1
session.index+numBytes  2
ret  [254]
session.index  2
session.index  2
numBytes  1
session.index+numBytes  3
ret  [2]
session.index  3
session.index  3
numBytes  2
session.index+numBytes  5
ret  [127 254]
session.index  5
session.index  5
numBytes  32766
session.index+numBytes  32771
erro
panic: runtime error: slice bounds out of range [:32771] with capacity 20480

**Comment by sijms** [comment, marked as answer]
I answer in issue #94
